ft_denoise_ssp.m

```diff
@@ -78,16 +100,45 @@
 data = ft_checkdata(data, 'datatype', 'raw', 'feedback', 'yes', 'hassampleinfo', 'yes');
 
 % check whether it is neuromag data
-if ~ft_senstype(data, 'neuromag')
-  ft_error('SSP vectors can only be applied to neuromag data');
+if isempty(refdata) && ~ft_senstype(data, 'neuromag')
+  ft_error('SSP without reference data can only be applied to neuromag data');
 end
 
 % select trials of interest
-tmpcfg = keepfields(cfg, {'trials', 'showcallinfo', 'trackcallinfo', 'trackusage', 'trackdatainfo', 'trackmeminfo', 'tracktimeinfo', 'checksize'});
+tmpcfg = keepfields(cfg, {'channel', 'trials', 'showcallinfo', 'trackcallinfo', 'trackusage', 'trackdatainfo', 'trackmeminfo', 'tracktimeinfo', 'checksize'});
 data   = ft_selectdata(tmpcfg, data);
+
+if ~isempty(refdata)
+    datachs = ft_channelselection('MEG',data.label); % pick MEG channels from data
+    refdata = ft_checkdata(refdata, 'datatype', 'raw', 'feedback', 'yes', 'hassampleinfo', 'yes');
+    if ~isempty(setdiff(datachs,refdata.label))
+        ft_error('Refdata is missing channels present in data.')
+    elseif ~isempty(setdiff(refdata.label,datachs))
+        ft_warning('Data is missing channels present in refdata. Dropping extra channels from refdata.')
+    end
+    cfg.channel = datachs;  
+    tmpcfg = keepfields(cfg, {'channel', 'showcallinfo', 'trackcallinfo', 'trackusage', 'trackdatainfo', 'trackmeminfo', 'tracktimeinfo', 'checksize'});
+    refdata = ft_selectdata(tmpcfg, refdata);
+end
+
 % restore the provenance information
 [cfg, data] = rollback_provenance(cfg, data);
 
+if ~isempty(refdata)

```

Comment:
should we consider to have ft_componentanalysis to do the heavy lifting here? One motivation could be that ft_componentanalysis is NaN aware, not sure about pca.